Data/Aeson/Text.hs

```diff
@@ -26,7 +27,9 @@ import Prelude.Compat
 
 import Data.Aeson.Types (Value(..), ToJSON(..))
 import Data.Aeson.Encoding (encodingToLazyByteString)
+#if !MIN_VERSION_base(4,11,0)
```

Comment:
I don't like this. We have `NoImplicitPrelude` and `base-compat` / `Prelude.Compat` exactly because we don't want (enable and) copy CPP to every file.

I'd say we should then rather add `other-modules: Data.Aeson.Prelude` and use it, to have CPP in one place only.

Reply:
Wonderful, let's do that instead.

Reply:
we can import Data.Semigroup.Compat instead

Reply:
The already imported `Prelude.Compat` seemed to do the job.